tools/tests/test_maskfile_configuration.py

```
@@ -0,0 +1,236 @@
+        content = _read_maskfile()
+        ci_section_match = re.search(r"#### ci\n.*?```bash\n(.*?)```", content, re.DOTALL)
+        assert ci_section_match is not None
+        ci_code = ci_section_match.group(1)
+        assert "cargo test" in ci_code, "CI must run tests"
+
+    def test_all_code_blocks_have_set_euo_pipefail(self) -> None:
+        content = _read_maskfile()
+        code_blocks = _extract_code_blocks(content)
+        for block in code_blocks:
+            assert "set -euo pipefail" in block, \
+                f"All bash blocks must start with 'set -euo pipefail', found block without it"
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)